pkg/cloudclient/aws/aws.go

```diff
@@ -77,72 +75,14 @@ func (c *Client) SetDefaultAPIPublic(ctx context.Context, kclient client.Client,
 
 // Healthcheck performs basic calls to make sure client is healthy
 func (c *Client) Healthcheck(ctx context.Context, kclient client.Client) error {
-	// Build the load balancer tag to look for.
-	clusterName, err := baseutils.GetClusterName(kclient)
-	if err != nil {
-		return err
-	}
-	var rhApiTags []*elb.Tag
-	rhApiTags = append(rhApiTags, &elb.Tag{
-		Key:   aws.String("kubernetes.io/cluster/" + clusterName),
-		Value: aws.String("owned"),
-	})
-	rhApiTags = append(rhApiTags, &elb.Tag{
-		Key:   aws.String("kubernetes.io/service-name"),
-		Value: aws.String("openshift-kube-apiserver/rh-api"),
-	})
-
 	elbList := make([]*elb.LoadBalancerDescription, 0)
 	input := &elb.DescribeLoadBalancersInput{}
-	err = c.elbClient.DescribeLoadBalancersPages(input, func(page *elb.DescribeLoadBalancersOutput, lastPage bool) bool {
+	err := c.elbClient.DescribeLoadBalancersPages(input, func(page *elb.DescribeLoadBalancersOutput, lastPage bool) bool {
 		elbList = append(elbList, page.LoadBalancerDescriptions...)
 		return lastPage
 	})
-	if err != nil {
-		return err
-	}
-
-	names := []*string{}
-	for _, lb := range elbList {
-		names = append(names, lb.LoadBalancerName)
-	}
-
-	// Request tags for up to 20 load balancers at a time.
-	for i := 0; i < len(names); i += 20 {
-		end := i + 20
-		if end > len(names) {
-			end = len(names)
-		}
-		tagsOutput, err := c.elbClient.DescribeTags(
-			&elb.DescribeTagsInput{
-				LoadBalancerNames: names[i:end],
-			},
-		)
-		if err != nil {
-			return err
-		}
-
-		for _, tag := range tagsOutput.TagDescriptions {
-			if includes(tag.Tags, rhApiTags) {
```

Comment:
My concern is that it would also hide bugs. 
Overall, I believe the healthcheck should focus on the technical readiness of the operator (able to call AWS, SA properly set, ...), and we should then implement monitoring on reconcile faillure/rh-api ELB not present in order to get alerted on the specific issue if something is not working (and investigate/fix it then).

Reply:
agree +1 on not putting complex logic into healthchecks